providers/gitlab/resources/gitlab.go

```diff
@@ -228,12 +228,13 @@ func (p *mqlGitlabProject) protectedBranches() ([]interface{}, error) {
 		isDefaultBranch := branch.Name == defaultBranch
 
 		branchSettings := map[string]*llx.RawData{
-			"name":           llx.StringData(branch.Name),
-			"allowForcePush": llx.BoolData(branch.AllowForcePush),
-			"defaultBranch":  llx.BoolData(isDefaultBranch),
+			"name":              llx.StringData(branch.Name),
+			"allowForcePush":    llx.BoolData(branch.AllowForcePush),
+			"defaultBranch":     llx.BoolData(isDefaultBranch),
+			"codeOwnerApproval": llx.BoolData(branch.CodeOwnerApprovalRequired),
 		}
 
-		mqlProtectedBranch, err := CreateResource(p.MqlRuntime, "gitlab.project.repository.protectedBranch", branchSettings)
+		mqlProtectedBranch, err := CreateResource(p.MqlRuntime, "gitlab.project.protectedBranch", branchSettings)
```

Comment:
it looks like the id function for this still needs to be updated for the new naming

Reply:
Thanks, Its correct:

func (g *mqlGitlabProjectProtectedBranch) id() (string, error) {
	return g.Name.Data, nil
}